src/controller.ts

```diff
@@ -426,44 +427,74 @@ export class Controller extends TSEventEmitter<Events> {
-		options = options || {
-			detached: true,
-			env: process.env,
-			stdio: 'ignore',
+		let joltronLogs = true;
+		let joltronOut: { name: string, fd: number } | null = null;
+		let joltronErr: { name: string, fd: number } | null = null;
+		let joltronOutLogger: Tail | null = null;
+		let joltronErrLogger: Tail | null = null;
 
-			// Uncomment to debug joltron output
-			// stdio: ['ignore', _fs.openSync('joltron.log', 'a'), _fs.openSync('joltron.log', 'a')],
-		};
+		try {
+			joltronOut = await fs.tmpFile('joltron-', 'out');
+			joltronErr = await fs.tmpFile('joltron-', 'err');
+			joltronOutLogger = Logger.createLoggerFromFile(joltronOut.name, 'joltron', 'info');
+			joltronErrLogger = Logger.createLoggerFromFile(joltronErr.name, 'joltron', 'error');
+			console.log(`Logging joltron output to "${joltronOut.name}" and "${joltronErr.name}"`);
+		} catch (e) {
+			console.warn('Failed to make temp log files for joltron. Logs from joltron will be disabled:', e);
+			joltronLogs = false;
+		}
 
-		let runnerExecutable = getExecutable();
+		try {
+			options = options || {
+				detached: true,
+				env: process.env,
+				stdio: joltronLogs ? ['ignore', joltronOut.fd, joltronErr.fd] : 'ignore',
+			};
 
-		// Ensure that the runner is executable.
-		await fs.chmod(runnerExecutable, '0755');
+			let runnerExecutable = getExecutable();
 
-		const portArg = args.indexOf('--port');
-		if (portArg === -1) {
-			throw new Error(`Can't launch a new instance without specifying a port number`);
-		}
-		const port = parseInt(args[portArg + 1], 10);
+			// Ensure that the runner is executable.
+			await fs.chmod(runnerExecutable, '0755');
+
+			const portArg = args.indexOf('--port');
+			if (portArg === -1) {
+				throw new Error(`Can't launch a new instance without specifying a port number`);
+			}
+			const port = parseInt(args[portArg + 1], 10);
 
-		console.log('Spawning ' + runnerExecutable + ' "' + args.join('" "') + '"');
-		const runnerProc = cp.spawn(runnerExecutable, args, options);
-		runnerProc.unref();
+			console.log('Spawning ' + runnerExecutable + ' "' + args.join('" "') + '"');
+			const runnerProc = cp.spawn(runnerExecutable, args, options);
+			runnerProc.unref();
 
-		const runnerInstance = new Controller(port, {
-			process: runnerProc.pid,
-			keepConnected: !!options.keepConnected,
-		});
-		runnerInstance.connect();
-		return runnerInstance;
-
-		// try {
-		// 	await runnerInstance.connect();
-		// 	return runnerInstance;
-		// } catch (err) {
-		// 	await runnerInstance.kill();
-		// 	throw err;
-		// }
+			const runnerInstance = new Controller(port, {
+				process: runnerProc.pid,
+				keepConnected: !!options.keepConnected,
+			});
+			runnerInstance.connect();
+			runnerInstance.on('close', () => {
+				if (joltronLogs) {
+					joltronOutLogger.unwatch();
+					joltronErrLogger.unwatch();
+					joltronLogs = false;
+				}
+			})
+			return runnerInstance;
+
+			// try {
```

Comment:
Remove Comment? If we want to keep it, add a comment explaining why.

Reply:
Ah, I didn't add this comment in this PR so I wanted to not touch it for minimal diffs. I don't mind removing it.